test/mocha/unit/rpc/FallbackHandler.test.ts

```diff
@@ -0,0 +1,151 @@
+              "returnData": true
+            },
+            {
+              "id": "callSuccesses",
+              "metricStat": {
+                "metric": {
+                  "namespace": "Uniswap",
+                  "name": "RPC_GATEWAY_56_QUIKNODE_SUCCESS",
+                  "dimensions": {
+                    "Service": "RoutingAPI"
+                  }
+                },
+                "period": 300,
+                "stat": "Sum",
+                "unit": "Count"
+              },
+              "returnData": false
+            }
+          ]
+        }
+      }
+    }`)
+  const recoverEvent = JSON.parse(`
+    {
+      "source": "aws.cloudwatch",
+      "alarmArn": "arn:aws:cloudwatch:us-east-2:901338192186:alarm:RoutingAPI-RpcGateway-ErrorRateAlarm-ChainId-56-Provider-QUIKNODE",
+      "accountId": "901338192186",
+      "time": "2024-04-16T08:47:36.960+0000",
+      "region": "us-east-2",
+      "alarmData": {
+        "alarmName": "RoutingAPI-RpcGateway-ErrorRateAlarm-ChainId-56-Provider-QUIKNODE",
+        "state": {
+          "value": "OK",
+          "reason": "Threshold Crossed: 1 datapoint [95.0 (16/04/24 08:33:00)] was not greater than or equal to the threshold (150.0).",
+          "reasonData": "{\\"version\\":\\"1.0\\",\\"queryDate\\":\\"2024-04-16T08:43:36.957+0000\\",\\"startDate\\":\\"2024-04-16T08:33:00.000+0000\\",\\"period\\":300,\\"recentDatapoints\\":[95.0],\\"threshold\\":150.0,\\"evaluatedDatapoints\\":[{\\"timestamp\\":\\"2024-04-16T08:33:00.000+0000\\",\\"value\\":95.0}]}",
+          "timestamp": "2024-04-16T08:47:36.960+0000"
+        },
+        "previousState": {
+          "value": "ALARM",
+          "reason": "Threshold Crossed: 1 datapoint [161.0 (16/04/24 08:42:00)] was greater than or equal to the threshold (150.0).",
+          "reasonData": "{\\"version\\":\\"1.0\\",\\"queryDate\\":\\"2024-04-16T08:47:36.958+0000\\",\\"startDate\\":\\"2024-04-16T08:42:00.000+0000\\",\\"period\\":300,\\"recentDatapoints\\":[161.0],\\"threshold\\":150.0,\\"evaluatedDatapoints\\":[{\\"timestamp\\":\\"2024-04-16T08:42:00.000+0000\\",\\"value\\":161.0}]}",
+          "timestamp": "2024-04-16T08:43:36.960+0000"
+        },
+        "configuration": {
+          "metrics": [
+            {
+              "id": "expr_1",
+              "expression": "callSuccesses",
+              "returnData": true
+            },
+            {
+              "id": "callSuccesses",
+              "metricStat": {
+                "metric": {
+                  "namespace": "Uniswap",
+                  "name": "RPC_GATEWAY_56_QUIKNODE_SUCCESS",
+                  "dimensions": {
+                    "Service": "RoutingAPI"
+                  }
+                },
+                "period": 300,
+                "stat": "Sum",
+                "unit": "Count"
+              },
+              "returnData": false
+            }
+          ]
+        }
+      }
+    }`)
+
+  it('test readAlarmEvent', async () => {
```

Comment:
the INSUFFICIENT DATA transitions are not tested.
You have
OK -> ALARM
ALARM -> OK

We also need
UD -> OK
UD -> ALARM
ALARM -> UD
OK -> UD

Reply:
UT for all these cases are added.